
This patch introduces the $XDG_CHECK_INTERVAL environment variable
which tells xdgmime how frequently it can recheck its data.
Setting the variable to a negative number disables rechecks.
This is useful if you're running your program under a slow
debugger, like duma.

diff --git a/gio/xdgmime/xdgmime.c b/gio/xdgmime/xdgmime.c
index 0888583..5c758be 100644
--- a/gio/xdgmime/xdgmime.c
+++ b/gio/xdgmime/xdgmime.c
@@ -418,6 +418,8 @@ xdg_check_dirs (void)
 static int
 xdg_check_time_and_dirs (void)
 {
+  static int check_interval_checked = 0;
+  static int check_interval = 5;
   struct timeval tv;
   time_t current_time;
   int retval = FALSE;
@@ -425,7 +427,19 @@ xdg_check_time_and_dirs (void)
   gettimeofday (&tv, NULL);
   current_time = tv.tv_sec;
 
-  if (current_time >= last_stat_time + 5)
+  if (!check_interval_checked)
+    {
+      const char *env;
+
+      env = getenv ("XDG_CHECK_INTERVAL");
+      if (env != NULL)
+        check_interval = atoi(env);
+      check_interval_checked = 1;
+    }
+
+  /* Don't recheck the dirs if @check_interval is negative. */
+  if (current_time >= last_stat_time + check_interval
+      && (!last_stat_time || check_interval > 0))
     {
       retval = xdg_check_dirs ();
       last_stat_time = current_time;
